pkg/cli/admin/release/image_mapper.go

```diff
@@ -435,7 +515,10 @@ func parseComponentVersionsLabel(label string) (ComponentVersions, error) {
 			return nil, fmt.Errorf("the version pair %q must have a valid semantic version: %v", pair, err)
 		}
 		v.Build = nil
-		labels[parts[0]] = v.String()
+		labels[parts[0]] = ComponentVersion{
```

Comment:
nit: Do we need to worry about having the same version name listed twice causing the last found to take precedence? I tend to think just noting "don't do that please" is enough.

Reply:
there's logic in the build that errors if that happens (if you try to create a release payload that has image A with component foo @ 1.2.0 and image B with component foo @1.2.1 you can't build the payload).  The logic is copied for display name, see the tests I added that use an image stream.

Reply:
Test is https://github.com/openshift/oc/pull/209/files#diff-a42d49db6f7580ec9dea5398fe27b8deR62